Gruntfile.js

```diff
@@ -0,0 +1,1259 @@
+      },
+      ccaReleaseAndroid: {
+        cwd: '<%= androidDistPath %>',
+        command: '<%= ccaJsPath %> build android --release --webview=crosswalk@org.xwalk:xwalk_core_library_beta:20.50.533.6'
+      },
+      ccaEmulateAndroid: {
+        cwd: '<%= androidDevPath %>',
+        command: '<%= ccaJsPath %> run android --emulator'
+      },
+      ccaCreateIosDev: {
+        command: '<%= ccaJsPath %> create <%= iosDevPath %> org.uproxy.uProxy "uProxy" --link-to=<%= ccaDevPath %>'
+      },
+      ccaCreateIosDist: {
+        command: '<%= ccaJsPath %> create <%= iosDistPath %> org.uproxy.uProxy "uProxy" --link-to=<%= ccaDevPath %>'
+      },
+      ccaAddPluginsIosBuild: {
+        cwd: '<%= iosDevPath %>',
+        command: '<%= ccaAddPluginsIosCmd %>'
+      },
+      addIosrtcHook: {
+        cwd: '<%= iosDevPath %>',
+        command: '<%= addIosrtcHookCmd %>'
+      },
+      ccaPrepareIosDev: {
+        cwd: '<%= iosDevPath %>',
+        command: '<%= ccaJsPath %> prepare'
+      },
+      ccaPrepareIosDist: {
+        cwd: '<%= iosDistPath %>',
+        command: '<%= ccaJsPath %> prepare'
+      },
+      cleanAndroid: {
+        command: 'rm -rf <%= androidDevPath %>; rm -rf <%= androidDistPath %>'
+      },
+      cleanIos: {
+        command: 'rm -rf <%= iosDevPath %>; rm -rf <%= iosDistPath %>'
+      },
+      androidReplaceXwalkDev: {
+        command: './replace_xwalk_in_apk.sh debug'
+      },
+      androidReplaceXwalkDist: {
+        command: './replace_xwalk_in_apk.sh release'
+      },
+      installFreedomForNodeForZork: {
+        // This allows our Docker containers, which do not have access to the
+        // git repo's "top-level" node_modules/ folder find freedom-for-node.
+
+        command: 'npm install --prefix build/src/lib/samples/zork-node freedom-for-node'
+      }
+    },
+    copy: {
+      // Copy releveant non-typescript src files to dev build.
+      resources: {
+        files: [
+          {
+            nonull: true,
+            expand: true,
+            cwd: 'src/',
+            src: ['**/*', '!**/*.ts', '!generic_core/dist_build/*', '!generic_core/dev_build/*'],
+            dest: devBuildPath,
+            onlyIf: 'modified'
+          }
+        ]
+      },
+      devGenericCore: {
+        files: [
+          {
+            nonull: true,
+            src: 'src/generic_core/dev_build/freedom-module.json',
+            dest: devBuildPath + '/generic_core/freedom-module.json',
+            onlyIf: 'modified'
+          }
+        ]
+      },
+      // Copy releveant files for distribution.
+      dist: {
+        files: [
+          {  // Chrome extension
+            expand: true,
+            cwd: chromeExtDevPath,
+            src: ['manifest.json', '_locales/**', 'generic_ui/scripts/copypaste.js', 'scripts/context.static.js', 'scripts/background.static.js'].concat(uiDistFiles, universalDistFiles),
```

Comment:
This is very long - why not have them on separate lines? In fact, generally, should we just stick to newlines after brackets and parentheses? I'm happy for that to appear in a followup, though, given the trickiness of merging this.